### PR TITLE
Allow moving global variables to separate modules.

### PR DESCRIPTION
**Pull request by alex4o**
This it turns out is not enough, but it is sufficient enough, at the moment.

### CODE REVIEW

**Comment by ltratt** [comment]
> This it turns out is not enough, but it is sufficient enough, at the moment.

Can you spell out why it isn't enough?

**Comment by alex4o** [comment]
Global Variables can depend on other global variables, and at the moment I don't check that. And also I might have to modify the visibility of the global variables for some programs.

**Comment by ltratt** [comment]
OK. Would it be possible to add small, artificial, test programs to this repo that test the splitter as it evolves?

**Comment by alex4o** [comment]
I have a test, but I can add a few more, that show how this works.

**Comment by ltratt** [comment]
I think it might be nice to add a series of gradually more complex examples as each PR increases the functionality.

**Comment by alex4o** [comment]
Yes I will add more examples.

**Comment by ltratt** [comment]
Thanks!

**Comment by alex4o** [comment]
Ok, here is a longer explanation. My testing was flawed, and now I kind of see what exactly I have to do to finish this thing. And it might not be too complicated. The problem was that llvm-extract doesn't make the extracted values public and the thing doesn't want to link properly.

**Comment by ltratt** [comment]
I'm fine with doing it in stages. A small PR with some tests now, and then a second PR which does even better (with more tests!) is probably a good way of doing it. We don't have to do everything at once :)

**Comment by alex4o** [comment]
Ok, then I will make a commit with the test files, I am using currently.

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
It turns out the problem wasn't the shell at all. It was the fact that I was spawning processes and not waiting for them.

**Comment by ltratt** [comment]
Can we not just remove the parallel code? It doesn't look very reliable at the moment to me.

**Comment by ltratt** [comment]
[In other words, adding `&` isn't really the right way to do parallelism. I'd like to do a proper job, but not in is PR.]

**Comment by alex4o** [comment]
I put it there because It works for me, and I use it to experiment with cpython, otherwise I have to wait 30min to finish splitting.

**Comment by ltratt** [comment]
We've already seen it bite us. Let's remove it from this PR and in the next PR we can do it properly, where we spawn and `wait` upon multiple child processes.

**Comment by alex4o** [comment]
bors try

**Comment by ltratt** [comment]
Assuming you're OK with this now, please squash.

**Comment by alex4o** [comment]
done

File: .buildbot.sh [hunk 1]
```diff
@@ -4,10 +4,18 @@ set -e
...
-make CC=$CHERI/bin/clang CXX=$CHERI/bin/clang++ LLVM_CONFIG=$CHERI/bin/llvm-config
-make bitcode CC=$CHERI/bin/clang CXX=$CHERI/bin/clang++ 
-make split CC=$CHERI/bin/clang CXX=$CHERI/bin/clang++ LLVM_CONFIG=$CHERI/bin/llvm-config
+make CC=$CC CXX=$CHERI/bin/clang++ LLVM_CONFIG=$CHERI/bin/llvm-config
+make split CC=$CC CXX=$CHERI/bin/clang++ LLVM_CONFIG=$CHERI/bin/llvm-config
 
-mkdir -p out
-./split file.bc -o ./out
+make test-included CC=$CC CXX=$CHERI/bin/clang++ 
+
```
Comment (ltratt) [inline .buildbot.sh:13]: Can you force push an update that removes the extra blank line (unless there's a good reason for it, but I think it's just a typo)?

Comment (alex4o) [reply .buildbot.sh:13]: It wasn't, I use it to logically separate the different tests. But I will remove it.

**Comment by alex4o** [comment]
Done

**Pull request merged**